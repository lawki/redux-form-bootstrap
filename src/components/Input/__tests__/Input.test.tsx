import * as React from "react";
import { shallow } from "enzyme";
import { Input as RSInput } from 'reactstrap';
import { InputComponent as Input } from "..";

describe("Input Component", () => {
  it('should render correctly', () => {
    const fn = jest.fn();
    const component = shallow(<Input type="text" onChange={fn} value="1"/>);
    expect(component).toMatchSnapshot();
  });

  it('has only one instance of reactstrap\'s Input ', () => {
    const fn = jest.fn();
    const app = shallow(<Input type="text" onChange={fn}/>);
    expect(app.find(RSInput).length).toBe(1);
  });

  it('calls change event handler', ()=>{
    const fn = jest.fn();
    const app = shallow(<Input type="text" onChange={fn}/>);
    app.find(RSInput).simulate('change', { target: { value: 30} });
    expect(fn.call.length).toBe(1);
  });

  it('set value correctly',()=>{
    const fn = jest.fn();
    const app = shallow(<Input type="text" onChange={fn} value={30}/>);
    app.find(RSInput).simulate('change', { target: { value: 30} });
    expect(fn.call.length).toBe(1);
  })
})
