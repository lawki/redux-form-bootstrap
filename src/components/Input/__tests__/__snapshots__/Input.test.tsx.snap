// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Component should render correctly 1`] = `
<Fragment>
  <InputGroup
    tag="div"
  >
    <Input
      onChange={[Function]}
      type="text"
      value="1"
    />
  </InputGroup>
</Fragment>
`;
